docs/madr/decisions/031-locality-aware-load-balancing.md

```diff
@@ -0,0 +1,380 @@
+
+![Use case 1](assets/031/use_case_1.png)
+
+#### Load balancing based on priorities within a zone. Example: node → AZ → region
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: ["k8s.io/node", "k8s.io/az"]
+```
+
+![Use case 2](assets/031/use_case_2_1.png)
+![Use case 2](assets/031/use_case_2_2.png)
+![Use case 2](assets/031/use_case_2_3.png)
+
+#### Prefer traffic in same zone with fallback to any other zone
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        fallback:
+          type: Any
+```
+
+![Use case 3](assets/031/use_case_3_1.png)
+![Use case 3](assets/031/use_case_3_2.png)
+
+#### Load balance equally to list of zones
+In this case, traffic from outside to the `backend` service in `zone-2` is completely excluded.
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - name: "group1"
+            zones: ["zone-1", "zone-3"]
+        fallback:
+          type: Fail
+```
+
+![Use case 4](assets/031/use_case_4_1.png)
+![Use case 4](assets/031/use_case_4_2.png)
+![Use case 4](assets/031/use_case_4_3.png)
+
+#### When leaving zone, target specific zones first
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1"]
+            name: "group1"
+        - groups:
+          - zones: ["zone-3"]
+            name: "group1"
+```
+
+![Use case 5](assets/031/use_case_5_1.png)
+![Use case 5](assets/031/use_case_5_2.png)
+![Use case 5](assets/031/use_case_5_3.png)
+
+#### Exclude single zone from load balancing
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        fallback:
+          type: Any
+          exclude:
+            zones: ["zone-3"]
+```
+
+
+![Use case 6](assets/031/use_case_6_1.png)
+![Use case 6](assets/031/use_case_6_2.png)
+![Use case 6](assets/031/use_case_6_3.png)
+
+#### Load balance equally within a group of zones
+
+When we want to keep traffic in specific groups and don't route fallback from EU to US.
+
+![Use case 8](assets/031/use_case_8.png)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - name: "us"
+            zones: ["us-east", "us-west"]
+          - name: "eu"
+            zones: ["eu-east"]
+        - groups:
+          - name: "eu"
+            zones: ["eu-east2"]
+        fallback:
+          type: Fail
+```
```

Comment:
I'm struggling to understand this example. Why Zone US does not fallback to eu-east2, but Zone EU does?

What is the relation between items in rules and items in groups?

Reply:
We can have many groups with different priorities, I assume that the user can define a mesh policy for all services differently than for others.

So we have a group with Priority 1, one in us and the other in eu. 
But the user has also another zone in eu and might want to fallback to `eu-east2` with priority 2 but fail for us.